sos/cleaner/parsers/hostname_parser.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.cleaner.parsers import SoSCleanerParser
+from sos.cleaner.mappings.hostname_map import SoSHostnameMap
+
+
+class SoSHostnameParser(SoSCleanerParser):
+
+    name = 'Hostname Parser'
+    map_file_key = 'hostname_map'
+    prep_map_file = 'sos_commands/host/hostname'
+    regex_patterns = [
+        r'(((\b|_)[a-zA-Z0-9-\.]{1,200}\.[a-zA-Z]{1,63}\b))'
+    ]
+
+    def __init__(self, conf_file=None, opt_domains=None):
+        super(SoSHostnameParser, self).__init__(conf_file)
+        self.mapping = SoSHostnameMap(opt_domains)
```

Comment:
These lines need to be swapped (and in other child classes of `SoSCleanerParser`). Since the `self.mapping` is updated in the `SoSCleanerParser` constructor (https://github.com/sosreport/sos/pull/2093/files#diff-48e63d79ded5dde8f6c63d408b0e48ffR52).

This causes the

    ERROR: Unable to obfuscate report: 'SoSHostnameParser' object has no attribute 'mapping'